src/main/java/parser/Parser.java

```diff
@@ -0,0 +1,298 @@
+package parser;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import exceptions.EmptyTaskException;
+import exceptions.InvalidInputException;
+import exceptions.TaskIndexOutOfBound;
+import storage.Storage;
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.TaskList;
+import task.Todo;
+
+/**
+ * The Parser class is responsible for parsing and executing user commands.
+ * It interprets the user's input and interacts with the TaskList, Ui, and Storage
+ * classes to perform the appropriate actions.
+ */
+public class Parser {
+
+    /**
+     * Parses and executes the user's command.
+     *
+     * @param command   The command entered by the user (e.g. Todo XXX).
+     * @param taskList  The TaskList object that holds all the tasks.
+     * @param storage   The Storage object that handles saving and loading tasks from the file.
+     * @return The result of the command as a string message.
+     * @throws InvalidInputException if the user enters an unrecognized command or provides invalid input.
+     * @throws EmptyTaskException    if the user attempts to add a task without providing a description.
+     * @throws TaskIndexOutOfBound   if the user provides an index for a task that does not exist.
+     */
+    public static String parseCommand(
+            String command, TaskList taskList, Storage storage
+    ) throws InvalidInputException, EmptyTaskException, TaskIndexOutOfBound {
+        String[] slicedStr = command.split(" ");
```

Comment:
I feel that the `slicedStr` could be renamed to something like `commandParts` or `inputParts` to emphasise that it's the user command + arguments being broken down into different parts for processing purposes.